Mathlib/CategoryTheory/Limits/Preserves/Presheaf.lean

```diff
@@ -0,0 +1,178 @@
+@[simps!]
+def functorToInterchangeIso : functorToInterchange A K ≅
+    K ⋙ (coyoneda ⋙ (whiskeringLeft _ _ _).obj (CostructuredArrow.proj _ _)) :=
+  NatIso.ofComponents (fun _ => Iso.refl _)
+
+/-- (Implementation) One way to express the flipped version of our functor. -/
+@[simps!]
+def flipFunctorToInterchange : (functorToInterchange A K).flip ≅
+    ((CostructuredArrow.proj yoneda A ⋙ yoneda) ⋙ (whiskeringLeft J Cᵒᵖ (Type u)).obj K) :=
+  NatIso.ofComponents (fun _ => Iso.refl _)
+
+/-- (Implementation) A natural isomorphism we will need to construct `iso`. -/
+@[simps! (config := { fullyApplied := false }) hom_app]
+noncomputable def isoAux :
+  (CostructuredArrow.proj yoneda A ⋙ yoneda ⋙ (evaluation Cᵒᵖ (Type u)).obj (limit K)) ≅
+    ((coyoneda ⋙ (whiskeringLeft (CostructuredArrow yoneda A) C (Type u)).obj
+      (CostructuredArrow.proj yoneda A)).obj (limit K)) :=
+  NatIso.ofComponents (fun _ => Iso.refl _)
+
+/-- (Implementation) The isomorphism that proves that `A` preserves finite limits. -/
+noncomputable def iso [IsFiltered (CostructuredArrow yoneda A)] :
+    A.obj (limit K) ≅ limit (K ⋙ A) := calc
+  A.obj (limit K) ≅ (colimit (CostructuredArrow.proj yoneda A ⋙ yoneda)).obj (limit K) :=
+      (IsColimit.coconePointUniqueUpToIso
+        (Presheaf.isColimitTautologicalCocone A) (colimit.isColimit _)).app _
+  _ ≅ colimit (((CostructuredArrow.proj yoneda A) ⋙ yoneda) ⋙ (evaluation _ _).obj (limit K)) :=
+      (colimitObjIsoColimitCompEvaluation _ _)
+  _ ≅ colimit ((CostructuredArrow.proj yoneda A) ⋙ yoneda ⋙ (evaluation _ _).obj (limit K)) :=
+      HasColimit.isoOfNatIso (Functor.associator _ _ _)
+  _ ≅ colimit
+      ((coyoneda ⋙ (whiskeringLeft _ _ _).obj (CostructuredArrow.proj yoneda A)).obj (limit K)) :=
+        HasColimit.isoOfNatIso (isoAux A K)
+  _ ≅ colimit (limit (K ⋙ (coyoneda ⋙ (whiskeringLeft _ _ _).obj (CostructuredArrow.proj _ _)))) :=
+        HasColimit.isoOfNatIso (preservesLimitIso _ _)
+  _ ≅ colimit (limit (functorToInterchange A K)) :=
+        HasColimit.isoOfNatIso (HasLimit.isoOfNatIso (functorToInterchangeIso A K).symm)
+  _ ≅ limit (colimit (functorToInterchange A K).flip) := colimitLimitIso _
+  _ ≅ limit (colimit
+        ((CostructuredArrow.proj yoneda A ⋙ yoneda) ⋙ (whiskeringLeft _ _ _).obj K)) :=
+          HasLimit.isoOfNatIso (HasColimit.isoOfNatIso (flipFunctorToInterchange A K))
+  _ ≅ limit (K ⋙ colimit (CostructuredArrow.proj yoneda A ⋙ yoneda)) :=
+        HasLimit.isoOfNatIso
+          (colimitCompWhiskeringLeftIsoCompColimit (CostructuredArrow.proj yoneda A ⋙ yoneda) K)
+  _ ≅ limit (K ⋙ A) := HasLimit.isoOfNatIso (isoWhiskerLeft _
+        (IsColimit.coconePointUniqueUpToIso
+          (colimit.isColimit _) (Presheaf.isColimitTautologicalCocone A)))
+
+theorem isIso_post [IsFiltered (CostructuredArrow yoneda A)] : IsIso (limit.post K A) := by
+  suffices limit.post K A = (iso A K).hom from this ▸ inferInstance
```

Comment:
I would think it would be nice to extract a lemma `iso_hom` from this proof.

Reply:
Done.